File: DFS/그래프 탐색.py
# 변수 선언 및 입력
n, m = map(int, input().split())

# 그래프를 딕셔너리로 선언
graph = {i: [] for i in range(1, n + 1)}
visited = [False] * (n + 1)
vertex_cnt = 0

def dfs(vertex):
    global vertex_cnt
    
    # 현재 정점을 방문 처리
    visited[vertex] = True
    vertex_cnt += 1
    
    # 연결된 모든 정점 탐색
    for neighbor in graph[vertex]:
        if not visited[neighbor]:
            dfs(neighbor)
    
# 간선 정보 입력
for _ in range(m):
    v1, v2 = map(int, input().split())
    graph[v1].append(v2)
    graph[v2].append(v1)

# DFS 시작
dfs(1)

print(vertex_cnt - 1)  # 시작 정점을 제외한 방문한 정점의 수

File: jhyun/DFS/그래프_탐색.java
import java.util.*;
import java.io.*;

public class Main {
    static final int MAX_N = 1000;
    static int n, m;

    static List<Integer>[] graph = new ArrayList[MAX_N + 1];
    static boolean[] visited = new boolean[MAX_N + 1];
    static int vertexCnt = 0;

    static void DFS(int vertex) {
        for (int i = 0; i < graph[vertex].size(); i++) {
            int currV = graph[vertex].get(i);
            if (!visited[currV]) {
                visited[currV] = true;
                vertexCnt++;
                DFS(currV);
            }
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());

        for (int i = 1; i <= n; i++) {
            graph[i] = new ArrayList<>();
        }

        while (m-- > 0) {
            st = new StringTokenizer(br.readLine());
            int v1 = Integer.parseInt(st.nextToken());
            int v2 = Integer.parseInt(st.nextToken());

            graph[v1].add(v2);
            graph[v2].add(v1);
        }

        visited[1] = true;
        DFS(1);

        System.out.println(vertexCnt);
    }
}